test/src/state_machine/state.test.cpp

```diff
@@ -0,0 +1,781 @@
+                                        telemetry_data, sensors_data, motors_data);
+    hyped::state_machine::State *new_state = state->checkTransition(log);
+
+    if (has_emergency) {
+      ASSERT_EQ(new_state, FailureStopped::getInstance());
+    } else {
+      ASSERT_NE(new_state, FailureStopped::getInstance());
+    }
+  }
+}
+
+TEST_F(IdleTest, handlesCalibrateCommand)
+{
+  Data &data  = Data::getInstance();
+  Idle *state = Idle::getInstance();
+
+  EmergencyBrakes embrakes_data;
+  Navigation nav_data;
+  Batteries batteries_data;
+  Telemetry telemetry_data;
+  Sensors sensors_data;
+  Motors motors_data;
+
+  for (int i = 0; i < TEST_SIZE; i++) {
+    Randomiser::randomiseEmbrakes(embrakes_data);
+    Randomiser::randomiseNavigation(nav_data);
+    Randomiser::randomiseTelemetry(telemetry_data);
+    Randomiser::randomiseMotors(motors_data);
+
+    data.setEmergencyBrakesData(embrakes_data);
+    data.setNavigationData(nav_data);
+    data.setTelemetryData(telemetry_data);
+    data.setMotorData(motors_data);
+
+    bool has_emergency = checkEmergency(log, embrakes_data, nav_data, batteries_data,
+                                        telemetry_data, sensors_data, motors_data);
+
+    if (!has_emergency) {
+      bool calibrate_command                 = checkCalibrateCommand(log, telemetry_data);
+      hyped::state_machine::State *new_state = state->checkTransition(log);
+
+      if (!calibrate_command) { ASSERT_EQ(new_state, nullptr); }
+    }
+  }
+}
+
+TEST_F(IdleTest, handlesAllInitialised)
+{
+  Data &data  = Data::getInstance();
+  Idle *state = Idle::getInstance();
+
+  EmergencyBrakes embrakes_data;
+  Navigation nav_data;
+  Batteries batteries_data;
+  Telemetry telemetry_data;
+  Sensors sensors_data;
+  Motors motors_data;
+
+  for (int i = 0; i < TEST_SIZE; i++) {
+    Randomiser::randomiseEmbrakes(embrakes_data);
+    Randomiser::randomiseNavigation(nav_data);
+    Randomiser::randomiseTelemetry(telemetry_data);
+    Randomiser::randomiseMotors(motors_data);
+
+    data.setEmergencyBrakesData(embrakes_data);
+    data.setNavigationData(nav_data);
+    data.setTelemetryData(telemetry_data);
+    data.setMotorData(motors_data);
+
+    bool has_emergency = checkEmergency(log, embrakes_data, nav_data, batteries_data,
+                                        telemetry_data, sensors_data, motors_data);
+
+    bool calibrate_command = checkCalibrateCommand(log, telemetry_data);
+
+    if (!has_emergency && !calibrate_command) {
+      bool all_initialised = checkModulesInitialised(log, embrakes_data, nav_data, batteries_data,
+                                                     telemetry_data, sensors_data, motors_data);
+      hyped::state_machine::State *new_state = state->checkTransition(log);
+
+      if (all_initialised) {
+        ASSERT_EQ(new_state, FailureStopped::getInstance());
+      } else {
+        ASSERT_NE(new_state, FailureStopped::getInstance());
```

Comment:
This seems wrong to me. Surely we should make sure that in the case where there is no emergency, we have received the calibrating command and all modules are initialised we go into Calibrating.